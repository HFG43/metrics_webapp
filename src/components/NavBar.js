import { Link, useLocation } from 'react-router-dom';
import styles from './NavBar.module.css';
import micIcon from '../Assets/micIcon.svg';
import configIcon from '../Assets/configIcon.svg';

const NavBar = () => {
  const location = useLocation();
  const home = location.pathname === '/home';
  const navCentralText = home ? 'Market View' : 'TBD';

  return (
    <>
      <nav className={styles.navbar}>
        <div>
          <span>
            <Link to="/home">&lt;</Link>
          </span>
          <span>2022</span>
        </div>
        <p>{navCentralText}</p>
        <div className={styles.mic_config_container}>
          <img
            className={styles.navbar_mic_icon}
            src={micIcon}
            alt="microphone icon"
          />
          <img
            className={styles.navbar_config_icon}
            src={configIcon}
            alt="configuration icon"
          />
        </div>
      </nav>
    </>
  );
};

export default NavBar;
