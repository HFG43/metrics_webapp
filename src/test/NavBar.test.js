import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import NavBar from '../components/NavBar';

test('The Navigation Bar renders correctly', () => {
  const tree = render(
    <MemoryRouter initialEntries={['/home']}>
      <NavBar />
    </MemoryRouter>,
  );
  expect(tree).toMatchSnapshot();
});

test('Check if the link is correctly placed in the component', () => {
  render(
    <MemoryRouter initialEntries={['/home']}>
      <NavBar />
    </MemoryRouter>,
  );
  const links = document.getElementsByTagName('a');
  expect(links.length).toBe(1);
});
