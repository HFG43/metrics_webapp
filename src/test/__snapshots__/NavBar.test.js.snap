// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Navigation Bar renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navbar_position"
      >
        <nav
          class="navbar"
        >
          <div>
            <span>
              <a
                href="/home"
              >
                &lt;
              </a>
            </span>
            <span>
              2022
            </span>
          </div>
          <p>
            Market View
          </p>
          <div
            class="mic_config_container"
          >
            <img
              alt="microphone icon"
              class="navbar_mic_icon"
              src="micIcon.svg"
            />
            <img
              alt="configuration icon"
              class="navbar_config_icon"
              src="configIcon.svg"
            />
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="navbar_position"
    >
      <nav
        class="navbar"
      >
        <div>
          <span>
            <a
              href="/home"
            >
              &lt;
            </a>
          </span>
          <span>
            2022
          </span>
        </div>
        <p>
          Market View
        </p>
        <div
          class="mic_config_container"
        >
          <img
            alt="microphone icon"
            class="navbar_mic_icon"
            src="micIcon.svg"
          />
          <img
            alt="configuration icon"
            class="navbar_config_icon"
            src="configIcon.svg"
          />
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
